components/raftstore/src/store/peer.rs

```diff
@@ -4033,12 +4040,17 @@ where
         ctx: &mut PollContext<EK, ER, T>,
         change_peer: CP,
         data: Vec<u8>,
+        is_unsafe_recovery_removing_peers: bool,
     ) -> Result<u64> {
         let data_size = data.len();
         let cc = change_peer.to_confchange(data);
         let changes = change_peer.get_change_peers();
 
-        self.check_conf_change(ctx, changes.as_ref(), &cc)?;
+        // Bypass the conf change check if it is a unsafe recovery removing peers request.
+        // It will be considered as a "Simple" v2 conf change.
+        if !is_unsafe_recovery_removing_peers {
+            self.check_conf_change(ctx, changes.as_ref(), &cc)?;
+        }
```

Comment:
I think some checks inside `check_conf_change` are still important, especially the `check_joint_state` check

Reply:
The `check_joint_state` check is port from [`apply_conf_change`](https://github.com/tikv/raft-rs/blob/master/src/raft.rs#L2736), but checking in advance before proposing instead of after applying, which is  because once `confchange` is applied, the result peer list is persisted hence can't revert, so usually tikv will panic if `apply_conf_change` return error

Reply:
Not applicable anymore